hamilton/plugins/h_polars.py

```diff
@@ -54,3 +67,152 @@ def build_result(
+    a, b, and a_plus_b to the dataframe. It then returns the dataframe, and does some processing on it."
+
+    In case you need more flexibility you can alternatively use ``pass_dataframe_as``, for example,
+
+    .. code-block:: python
+
+            # with_columns_module.py
+            def a_from_df(initial_df: pl.Series) -> pl.Expr:
+                return pl.col(a).alias("a") / 100
+
+            def b_from_df(initial_df: pl.Series) -> pd.Series:
+                return pl.col(a).alias("b") / 100
+
+
+            # the with_columns call
+            @with_columns(
+                *[my_module],
+                pass_dataframe_as="initial_df",
+                select=["a_from_df", "b_from_df", "a_plus_b", "a_b_average"],
+            )
+            def final_df(initial_df: pl.DataFrame) -> pl.DataFrame:
+                # process, or just return unprocessed
+                ...
+
+    the above would output a dataframe where the two columns ``a`` and ``b`` get
+    overwritten.
+    """
+
+    def __init__(
+        self,
+        *load_from: Union[Callable, ModuleType],
+        columns_to_pass: List[str] = None,
+        pass_dataframe_as: str = None,
+        select: List[str] = None,
+        namespace: str = None,
+        config_required: List[str] = None,
+    ):
+        """Instantiates a ``@with_columns`` decorator.
+
+        :param load_from: The functions or modules that will be used to generate the group of map operations.
+        :param columns_to_pass: The initial schema of the dataframe. This is used to determine which
+            upstream inputs should be taken from the dataframe, and which shouldn't. Note that, if this is
+            left empty (and external_inputs is as well), we will assume that all dependencies come
+            from the dataframe. This cannot be used in conjunction with pass_dataframe_as.
+        :param pass_dataframe_as: The name of the dataframe that we're modifying, as known to the subdag.
+            If you pass this in, you are responsible for extracting columns out. If not provided, you have
+            to pass columns_to_pass in, and we will extract the columns out for you.
+        :param select: The end nodes that represent columns to be appended to the original dataframe
+            via with_columns. The length of each column has to match the original dataframe length.
+            Existing columns will be overridden.
+        :param namespace: The namespace of the nodes, so they don't clash with the global namespace
+            and so this can be reused. If its left out, there will be no namespace (in which case you'll want
+            to be careful about repeating it/reusing the nodes in other parts of the DAG.)
+        :param config_required: the list of config keys that are required to resolve any functions. Pass in None\
+            if you want the functions/modules to have access to all possible config.
+        """
+
+        super().__init__(
```

Comment:
Curious, why are we delegating to the superclass here? Why not just pass through?